src/transformers/configuration_utils.py

```diff
@@ -313,6 +315,12 @@ def __init__(self, **kwargs):
                 logger.error(f"Can't set {key} with value {value} for {self}")
                 raise err
 
+        # Finally, store the original hash of the object (to detect post-loading changes)
+        self._original_object_hash = hash(self)
```

Comment:
This pattern is copied from GenerationConfig: store a hash of the object at load time to then be able to detect changes

NOTE: we won't be able to detect whether the user has changed generation parameters, only whether the object was changed in general. This simple detection should cover the base use case (reusing the config of a pretrained model), but results in a raised exception in more advanced usage (e.g. load a pretrained config with generative parameters set, and modify a non-generative parameter)